File: sdks/ruby/lib/novabilling/types/plan_override_response.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class PlanOverrideResponse < Internal::Types::Model
      field :id, -> { String }, optional: false, nullable: false
      field :customer_id, -> { String }, optional: false, nullable: false, api_name: "customerId"
      field :plan_id, -> { String }, optional: false, nullable: false, api_name: "planId"
      field :overridden_prices, -> { Internal::Types::Hash[String, Object] }, optional: true, nullable: false, api_name: "overriddenPrices"
      field :overridden_minimum_commitment, -> { Integer }, optional: true, nullable: false, api_name: "overriddenMinimumCommitment"
      field :overridden_charges, -> { Internal::Types::Hash[String, Object] }, optional: true, nullable: false, api_name: "overriddenCharges"
      field :metadata, -> { Internal::Types::Hash[String, Object] }, optional: true, nullable: false
      field :created_at, -> { String }, optional: false, nullable: false, api_name: "createdAt"
      field :updated_at, -> { String }, optional: false, nullable: false, api_name: "updatedAt"
    end
  end
end

File: sdks/csharp/src/NovaBilling/Types/PlanOverrideResponse.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record PlanOverrideResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("id")]
    public required string Id { get; set; }

    [JsonPropertyName("customerId")]
    public required string CustomerId { get; set; }

    [JsonPropertyName("planId")]
    public required string PlanId { get; set; }

    [JsonPropertyName("overriddenPrices")]
    public Dictionary<string, object?>? OverriddenPrices { get; set; }

    [JsonPropertyName("overriddenMinimumCommitment")]
    public double? OverriddenMinimumCommitment { get; set; }

    [JsonPropertyName("overriddenCharges")]
    public Dictionary<string, object?>? OverriddenCharges { get; set; }

    [JsonPropertyName("metadata")]
    public Dictionary<string, object?>? Metadata { get; set; }

    [JsonPropertyName("createdAt")]
    public required DateTime CreatedAt { get; set; }

    [JsonPropertyName("updatedAt")]
    public required DateTime UpdatedAt { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
